### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by mbarrenechea**
### Overview

This PR adds:
- `useAuth` hook to be able to access the user and functions everywhere thanks to a provider
- Passport + Next connect for implementing local strategy, API for saving a secure cookie in the client and deserialize it

### Testing instructions

Go to Marxan web and use the correct credentials

## Things missing
- [x] Improve signing, signup forms to handle errors.
- [x] Improve signing, signup visually.
- [x] Improve header behaviour

### CODE REVIEW

**Comment by hotzevzl** [comment]
@mbarrenechea we have added `displayName` as optional field for signups in an API PR ready for review, to follow the designs.

You won't see this in the OpenAPI docs on the dev instance until we approve/merge/deploy, so for reference the DTO is this one:
https://github.com/Vizzuality/marxan-cloud/blob/f4e8a4328c7770378d31ad81a4b765c006808ac5/api/src/modules/authentication/dto/sign-up.dto.ts

When editing users (no POST/PUT/DELETE endpoints for these yet - would you find it useful to have these sooner rather than later or are you happy to work with the other endpoints we already have? just let us know and we'll schedule accordingly), the API will allow to set other props such as (if needed) distinct first and last name, and a JSON `metadata` field (which in theory could be used as a generic K/V store, though since I am a nasty person I always prefer to *eventually* converge on versioned schemas to avoid sudden cosmic bangs due to random stuff creeping in JSON documents). We should also discuss what you would like to use for avatars. Sorry for hijacking an authn PR with user-related drivel, I guess we should move this to ~PT~ Jira 😬

**Comment by mbarrenechea** [comment]
@hotzevzl Ok, I didn't implement yet the user creation. Good to know in advance all the params involved.

**Comment by mbarrenechea** [comment]
@hotzevzl regarding the user edition, designers are working on this, so I'm not in a rush

File: app/pages/projects.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import Protected from 'layout/protected';
+
+const ProjectsPage: React.FC = () => {
+  const { user, errorRedirect } = useAuth();
+  const router = useRouter();
+
+  if (!user) {
+    router.push(errorRedirect);
+    return null;
+  }
```
Comment (clementprdhomme) [inline app/pages/projects.tsx:18]: This is already handled by the `Protected` component.

**Review by clementprdhomme — APPROVED** [review]
I reviewed the code and logged in/out of the platform too. The code is better than what I would have done 👍.

I added a few minor comments.

File: app/pages/projects.tsx (hunk 1 above)
Comment (mbarrenechea) [reply app/pages/projects.tsx]: Thanks for the catch

File: app/components/button/component.tsx [hunk 2]
```diff
@@ -30,15 +30,17 @@ export const Button: React.FC<ButtonProps> = ({
   theme = 'primary',
   size = 'base',
   className,
+  disabled,
```
[the remark this answers is not in the document]
Comment (mbarrenechea) [reply app/components/button/component.tsx:33]: 👍

**Comment by pablopareja** [comment]
I've had a look at the architecture and methods and it looks quite cool! 👏 
I've made a suggestion [here](https://github.com/Vizzuality/marxan-cloud/pull/56) to use preliminary interfaces for both the `User` object and the `AuthContextProps`
@mbarrenechea Let me know what you think ;)

**Pull request merged**